boa/src/builtins/array/mod.rs

```diff
@@ -936,26 +948,30 @@ impl Array {
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
     pub(crate) fn slice(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
         let new_array = Self::new_array(context)?;
-        let len = this.get_field("length").as_number().unwrap() as i32;
+        let len = this.get_field("length").to_length(context)? as isize;
 
         let start = match args.get(0) {
-            Some(v) => v.as_number().unwrap() as i32,
-            None => 0,
+            Some(v) => v.to_integer(context)?,
+            None => 0.0,
         };
         let end = match args.get(1) {
-            Some(v) => v.as_number().unwrap() as i32,
-            None => len,
+            Some(v) => v.to_integer(context)?,
+            None => len as f64,
         };
 
-        let from = if start < 0 {
-            max(len.wrapping_add(start), 0)
+        let from = if !start.is_finite() {
+            0
+        } else if start < 0.0 {
+            max(len.wrapping_add(start as isize), 0)
         } else {
-            min(start, len)
+            min(start as isize, len)
         };
-        let to = if end < 0 {
-            max(len.wrapping_add(end), 0)
+        let to = if !end.is_finite() {
+            0
+        } else if end < 0.0 {
+            max(len.wrapping_add(end as isize), 0)
         } else {
-            min(end, len)
+            min(end as isize, len)
```

Comment:
Another one that should be using `isize::try_from()`.

Reply:
@Razican I will have some time this evening (hopefully) and will go through the whole file, don't need to waste time searching for all the casts. :)